### DISCUSSION TITLE
Multiple squeezelite instances (in one Docker)

### DISCUSSION THREAD

**Discussion by stueble** [GioF71/squeezelite-docker#318] [Ideas]
Hi,

I am currently building a 24 channel amp based on three USB sound cards. The amp will be controlled by my home server running docker. Therefore, squeezelite-docker its a perfect basis for this. The only feature I am missing is to start multiple squeezelite instances in parallel. Any chance to support this feature or to include an appropriate merge request?

On possible implementation could be to allow users to add more sets of squeezelite configs, e.g., something such as "SQUEEZELITE_NAME_2" etc., or to allow users to define lists such as "SQUEEZELITE_NAME_2=stereo1,stereo2". But there may be better ways...

Best regards,
Chris

**Comment by GioF71** [comment]
Hello Chris, did you try to simply run multiple containers? As long as the server is specified with SQUEEZELITE_SERVER_PORT, all players should be recognized by the server and work. Without SQUEEZELITE_SERVER_PORT set, AFAIK only one instance would connect to the server. At least that's how LMS used to work, I didn't check with music assistant / home assistant, in case you are using that instead of LMS.

**Comment by GioF71** [comment]
If your concern is synchronizing the output, this should be done on the server. The fact that the players would run from the same container would not make them synchronized.

Of course tell me if I am missing something from your request.

**Comment by stueble** [comment]
Of course it is possible to start 12 squeezelite-docker containers to support 24 audio channels, but there are two problems here: Firstly, I need to map the sound card in all 12 containers and have to ensure multiplexing between them outside (e.g., using a mixer device in the host. Secondly, since only 8 channels together can be put into standby, the status of each 4 (stereo) to 8 (mono) squeezelite instances need to be synchronized. Moreover, I have one USB I/O board to mute the channels, which has to be controlled based on the players states. This is easier within one container. 

Maybe my use case may be too specifc and its easier if I overwrite your startup script with a hardcoded one.

**Comment by GioF71** [comment]
Hello, I probably don't understand your setup, specifically I don't know what you say about multiplexing and standby. Maybe you can map your asound.conf to each container if that is where you have created your configurations. You can map the asound.conf file by creating a volume like the following:

- ./asound.conf:/etc/asound.conf

assuming that there is a file names asound.conf where your compose file is. 

I don't know how running multiple instances of squeezelite in the same container can be beneficial, but of course as I said, maybe I am overlooking something due to lack of understanding.
Usually one container should contain one application, or, in particular cases, the required helper processes, while it's a bit unusual that a single container runs multiple instances of the main application.
